### PR TITLE
MCR-2137 improve mcr tile file provider

### PR DESCRIPTION
**Pull request by rsteph-de**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2137).

### CODE REVIEW

File: mycore-iview2/src/main/java/org/mycore/iview2/iiif/MCRIVIEWIIIFImageImpl.java [hunk 1]
```diff
@@ -289,5 +305,36 @@ private MCRTiledPictureProps getTiledPictureProps(Path tiledFile) throws MCRIIIF
...
+        return tileInfo;
+    }
+
+    private void checkTileFile(String identifier, MCRTileInfo tileInfo, Path tileFilePath)
+        throws MCRAccessException, MCRIIIFImageNotFoundException {
+        if (!Files.exists(tileFilePath)) {
+            throw new MCRIIIFImageNotFoundException(identifier);
+        }
+        if (tileInfo.getDerivate() != null) {
+            if (!MCRAccessManager.checkPermission(tileInfo.getDerivate(), MCRAccessManager.PERMISSION_READ)
```
Comment (adlerre) [inline mycore-iview2/src/main/java/org/mycore/iview2/iiif/MCRIVIEWIIIFImageImpl.java]: Codacy Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=4800529)

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- mycore-iview2/src/main/java/org/mycore/iview2/backend/MCRDefaultTileFileProvider.java  1
- mycore-iview2/src/main/java/org/mycore/iview2/backend/MCRTileInfo.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=4800529&bid=15975340)

File: mycore-iview2/src/main/java/org/mycore/iview2/backend/MCRTileInfo.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+package org.mycore.iview2.backend;
+
+/**
+ * Holds all attributes for a specific tile.
+ * @author Thomas Scheffler (yagee)
+ *
+ */
+public class MCRTileInfo {
+
+    String derivate, imagePath, tile;
```
Comment (adlerre) [inline mycore-iview2/src/main/java/org/mycore/iview2/backend/MCRTileInfo.java]: Codacy Issue found: [Use one line for each declaration, it enhances code readability.](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=4800529)

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-iview2/src/main/java/org/mycore/iview2/backend/MCRTileInfo.java  1
- mycore-iview2/src/main/java/org/mycore/iview2/backend/MCRDefaultTileFileProvider.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=4800529&bid=15975340)

File: mycore-user2/src/main/resources/xsl/user.xsl [hunk 3]
```diff
@@ -25,7 +25,7 @@
   <xsl:variable name="owns" select="document(concat('user:getOwnedUsers:',$uid))/owns" />
 
   <xsl:template match="user" mode="actions">
-    <xsl:variable name="isCurrentUser" select="$CurrentUser = /user/@name" />
+    <xsl:variable name="isCurrentUser" select="$CurrentUser = $uid" />
```
Comment (yagee-de) [inline mycore-user2/src/main/resources/xsl/user.xsl]: seems not to belong to the ticket

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- mycore-iview2/src/main/java/org/mycore/iview2/backend/MCRDefaultTileFileProvider.java  1
- mycore-iview2/src/main/java/org/mycore/iview2/frontend/MCRTileServlet.java  2
- mycore-iview2/src/main/java/org/mycore/iview2/backend/MCRTileInfo.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=4800529&bid=15975340)

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Complexity increasing per file
==============================
- mycore-iview2/src/main/java/org/mycore/iview2/backend/MCRDefaultTileFileProvider.java  1
- mycore-iview2/src/main/java/org/mycore/iview2/iiif/MCRIVIEWIIIFImageImpl.java  2
- mycore-iview2/src/main/java/org/mycore/iview2/backend/MCRTileInfo.java  1
- mycore-iview2/src/main/java/org/mycore/iview2/frontend/MCRTileServlet.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=4800529&bid=15975340)

**Review by kkrebs — APPROVED** [review]
successfully tested with MIR

🚢

**Review by Possommi — APPROVED** [review]
works with the collections app

**Pull request merged**